core/src/main/java/com/elefana/node/v2/V2OsStats.java

```diff
@@ -16,14 +16,140 @@
 package com.elefana.node.v2;
 
 import com.elefana.node.OsStats;
+import oshi.SystemInfo;
+import oshi.hardware.CentralProcessor;
+import oshi.hardware.GlobalMemory;
+import oshi.hardware.HardwareAbstractionLayer;
+import oshi.hardware.VirtualMemory;
 
+import java.util.Date;
+import java.util.HashMap;
 import java.util.Map;
 
 public class V2OsStats extends OsStats {
 
+	private static final long MEASURE_NOT_AVAILABLE = -1L;
+
+	private long[] recentCpuTicks;
+	private HardwareAbstractionLayer hardware = new SystemInfo().getHardware();
+	private CentralProcessor cpu = hardware.getProcessor();
+	private GlobalMemory memory = hardware.getMemory();
+	private VirtualMemory virtualMemory = memory.getVirtualMemory();
+
 	@Override
 	protected void generateCurrentStats(Map<String, Object> result) {
-		
+		result.clear();
+
+		updateCpu(result);
+		updateMemory(result);
+		updateSwap(result);
+		updateTimestamp(result);
+	}
+
+	private void updateCpu(Map<String, Object> osStatsObj) {
+		updateCpuCurrentLoad(osStatsObj);
+		updateCpuAverageLoad(osStatsObj);
+	}
+
+	private void updateCpuCurrentLoad(Map<String, Object> osStatsObj)
+	{
```

Comment:
Keep bracket placement consistent (in IntelliJ I think you can auto-format via Edit -> Advanced -> Format Document)